factory/declarations.py

```diff
@@ -100,24 +119,54 @@ def evaluate(self, instance, step, extra):
         return self.function(instance)
 
 
-class Transformer(LazyFunction):
-    """Transform value using given function.
+class _UNSPECIFIED:
+    pass
 
-    Attributes:
-        transform (function): returns the transformed value.
-        value: passed as the first argument to the transform function.
-    """
 
-    def __init__(self, transform, value, *args, **kwargs):
-        super().__init__(transform, *args, **kwargs)
-        self.value = value
+class Transformer(BaseDeclaration):
+    CAPTURE_OVERRIDES = True
+    UNROLL_CONTEXT_BEFORE_EVALUATION = False
 
-    def evaluate(self, instance, step, extra):
-        return self.function(self.value)
+    class Force:
+        """
+        Bypass a transformer's transformation.
 
+        The forced value can be any declaration, and will be evaluated as if it
+        had been passed instead of the Transformer declaration.
+        """
+        def __init__(self, forced_value):
+            self.forced_value = forced_value
 
-class _UNSPECIFIED:
-    pass
+        def __repr__(self):
+            return f'Transformer.Force({repr(self.forced_value)})'
+
+    def __init__(self, default, *, transform):
+        super().__init__()
+        self.default = default
+        self.transform = transform
+
+    def evaluate_pre(self, instance, step, overrides):
+        # The call-time value, if present, is set under the "" key.
+        value_or_declaration = overrides.pop("", self.default)
+
+        if isinstance(value_or_declaration, self.Force):
+            bypass_transform = True
+            value_or_declaration = value_or_declaration.forced_value
+        else:
+            bypass_transform = False
+
+        # We remove the wrapped declaration from the overrides dict,
+        # since the rest of the dict may be used when resolving the
+        # wrapped declaration.
```

Comment:
Looking back into this, I'm not sure I still understand what I meant with this comment :(

Reply:
Removed the comment during the rebase.